File: internal/entconfig/subject_max.go
package entconfig

// SubjectENTMax — максимум баллов ЕНТ по каждому предмету в тренажёре.
var SubjectENTMax = map[string]int{
	"История Казахстана":         20,
	"Математическая грамотность": 10,
	"Грамотность чтения":         10,
}

// MaxPoints returns the ENT max score for a subject.
func MaxPoints(subjectName string) int {
	if v, ok := SubjectENTMax[subjectName]; ok {
		return v
	}
	return 20
}

File: frontend/src/lib/subjectMax.js
/** ENT max per subject — must match internal/entconfig/subject_max.go */
export const SUBJECT_MAX = {
  'История Казахстана': 20,
  'Математическая грамотность': 10,
  'Грамотность чтения': 10,
}

export function subjectMaxScore(subjectName) {
  return SUBJECT_MAX[subjectName] ?? 20
}

export function predictionSubjectMax(prediction, subjectName) {
  return prediction?.max_score ?? subjectMaxScore(subjectName)
}
